src/index.ts

```diff
@@ -0,0 +1,566 @@
+    size: number
+    fee: number
+    modifiedfee: number
+    time: number
+    height: number
+    descendantcount: number
+    descendantsize: number
+    descendantfees: number
+    ancestorcount: number
+    ancestorsize: number
+    ancestorfees: number
+    wtxid: string
+    depends: string[]
+  }
+}
+
+export type DecodedRawTransaction = {
+  txid: string
+  hash: string
+  size: number
+  vsize: number
+  version: number
+  locktime: number
+  vin: TxIn[]
+  vout: TxOut[]
+}
+
+export interface FetchedRawTransaction extends DecodedRawTransaction {
+  hex: string
+  blockhash: string
+  confirmations: number
+  time: number
+  blocktime: number
+}
+
+export type MiningInfo = {
+  blocks: number
+  currentblockweight: number
+  currentblocktx: number
+  difficulty: number
+  networkhashps: number
+  pooledtx: number
+  chain: 'main' | 'test' | 'regtest'
+  warnings?: string
+}
+
+export type MempoolInfo = {
+  size: number
+  bytes: number
+  usage: number
+  maxmempol: number
+  mempoolminfee: number
+  minrelaytxfee: number
+}
+
+export type BlockHeader = {
+  hash: string
+  confirmations: number
+  height: number
+  version: number
+  versionHex: string
+  merkleroot: string
+  time: number
+  mediantime: number
+  nonce: number
+  bits: string
+  difficulty: number
+  chainwork: string
+  previoutsblockchash: string
+}
+
+export type Block = {
+  hash: string
+  confirmations: number
+  strippedsize: number
+  size: number
+  weight: number
+  height: number
+  version: number
+  verxionHex: string
+  merkleroot: string
+  tx: Transaction[] | string
+  hex: string
+  time: number
+  mediantime: number
+  nonce: number
+  bits: string
+  difficulty: number
+  chainwork: string
+  previousblockhash: string
+  nextblockchash?: string
+}
+
+export type Transaction = {
+  txid: string
+  hash: string
+  version: number
+  size: number
+  vsize: number
+  locktime: number
+  vin: TxIn[]
+  vout: TxOut[]
+}
+
+export type TxIn = {
+  txid: string
+  vout: number
+  scriptSig: {
+    asm: string
+    hex: string
+  }
+  txinwitness?: string[]
+  sequence: number
+}
+
+export type TxInForCreateRaw = {
+  txid: string
+  vout: number
+  sequence?: number
+}
+
+export type TxOut = {
+  value: number
+  n: number
+  scriptPubKey: {
+    asm: string
+    hex: string
+    reqSigs: number
+    type: scriptPubkeyType
+    addresses: string[]
+  }
+}
+
+export type TxOutForCreateRaw = {
+  address: string
+  data: string
+}
+
+export type TxOutInBlock = {
+  bestblock: string
+  confirmations: number
+  value: number
+  scriptPubKey: {
+    asm: string
+    hex: string
+    reqSigs: number
+    type: scriptPubkeyType
+    addresses: string[]
+  }
+  coinbase: boolean
+}
+
+export type WalletTransaction = {
+  amount: number
+  fee: number
+  confirmations: number
+  blockhash: string
+  blockindex: number
+  blocktime: number
+  txid: string
+  time: number
+  timereceived: number
+  'bip125-replaceable': 'yes' | 'no' | 'unknown'
```

Comment:
it is used in different places maybe create a type alias

Reply:
Done